### DISCUSSION TITLE
Building with verbose logging enabled

### DISCUSSION THREAD

**Discussion by midders3** [OpenVPN/openvpn3-linux#123] [Q&A - Our support arena] [answered]
Hi there

I'm a long time programmer, but new to github. Can anyone tell me what the preferred method is for building with verbose logging enabled in the openvpn client? I'm trying to troubleshoot a VPN connection that won't cooperate.

Slainte

midders3

**Comment by dsommers** [comment, marked as answer]
If you try to build `openvpn3-linux`, you can get a lot of debug info by boosting the log levels to `6`.  First, start with increasing the overall log level in the `openvpn3-service-logger`: 

     # openvpn3-admin log-service --log-level 6

Then, depending on how you start the VPN session there are more alternatives:

     $ openvpn2 --config CONFIG_FILE --verb 6

or edit the "CONFIG_FILE" to include `verb 6`

or import the configuration file and change the `log-level` setting on the imported profile:

     $ openvpn3 config-import --config CONFIG_FILE  --name myvpn
     $ openvpn3 config-manage --config myvpn --log-level 6
     $ openvpn3 session-start --config myvpn

or on an already running VPN session:

     $ openvpn3 session-manage --config CONFIG_NAME --log-level=6

And you can also retrieve run-time log events this way:

     $ openvpn3 log --config CONFIG_NAME


For building openvpn3-linux yourself, there are a few options worth looking into when running the `./configure` script.
* `--enable-debug-options` ... adds more options suitable for debugging
* `--enable-debug-core-events` ... Adds more low-level debug log events coming from the OpenVPN 3 Core library.

There are a few more, but they goes far more into the aspects of development debugging, not admin/user debugging.  You can anyhow see all of these options running `./configure --help`.

**Reply by midders3** [reply]
Thanks for the comprehensive response.

With the help of my VPN provider, eventually traced the problem to my ISP providing a restricted network (b*$&*%$s) and found a few ports that they don't restrict. Mentioning no names TalkTalk.
